### ISSUE TITLE
Unity indicators not displaying properly in systray

### ISSUE THREAD

**Issue by Dourak** [linuxmint/cinnamon#4035]
Hello,

I want to use 'indicator-multiload' to display resources usage in the panel.
It works fine with Mint 17 (Cinnamon 2.2 I believe), but with Mint 17.1 the graphs are shrinked to fit the width of a 'standard' icon, so they are unreadable.
The panel should allow icons to occupy the width they need.

Regards,
Thibaut Lepage

**Comment by lestcape** [comment]
@Dourak I think that the panel's functionality, try to create an standar way to display thing... when things are not standar, is not a fault of cinnamon panel, the fault are use the panel functionality to display the not standar things... Is possible override the panel functionality and create one according what you want... Is impossible control all possible behaviors.

Regards.

**Comment by dalcde** [comment]
Exactly which applet is causing the problem?

It is more likely to be a problem with the applet, so file this against the applet developer as well.

**Comment by Dourak** [comment]
indicator-multiload.
It needs enough horizontal space to display graphs.
Here is the position of the developer on the topic (see e.g. the last comment):
https://bugs.launchpad.net/indicator-multiload/+bug/1014494

**Comment by Paviluf** [comment]
P-sensor cannot display sensors infos on the systray whereas it works perfectly on Ubuntu

**Comment by JosephMcc** [comment]
Does this now work properly in Cinnamon 2.8?

**Comment by Dourak** [comment]
No, still not.




Date: Sun, 8 Nov 2015 02:47:07 -0800

**Comment by lestcape** [comment]
@Dourak and @JosephMcc, the appindicator implementation in cinnamon, only support one icon inside the panel. This mean, any other thing that indicator can be displayed will not be showed. I don't say will be impossible add more things, just is impossible right now see another thing.

Apparently indicators is more than a systray, so i think will be more effective create a different implementation where one indicator will convert to an special type of applet instead of a simple incon in the systray area. Thats ofcourse need to take in account that indicator can be moved one by one in a panel and all other standards features of an applet. In any case, this is certainly difficult to implement and will be necessary create an especial api to auto generate an applet from the abstract data provided by the Dbus interface.

**Comment by lestcape** [comment]
Anyway, this is how will look like psensor in cinnamon:
![psensor](https://cloud.githubusercontent.com/assets/1389572/11167248/004e412e-8b20-11e5-8742-65f7ce6319d8.png)
But only after merge this pull:
https://github.com/linuxmint/Cinnamon/pull/4814

**Comment by vladipus** [comment]
Multiload indicator is still not displayed correctly in Cinnamon 3.0 under Ubuntu 16.03 LTS.

**Comment by lestcape** [comment]
@vladipus Multiload indicator is displayed correctly here:
https://github.com/lestcape/Systray-Indicator

Who want can test the applet, as it's a normal applet, and find bugs with indicators... I will fix all that i can...

**Comment by lestcape** [comment]
The original report still exist in cinnamon 3.2 with and without indicator support.
For indicator i can make this happens, but will be a problem for vertical panels:
https://github.com/linuxmint/Cinnamon/pull/5790#issuecomment-254023638

**Comment by Vahan86** [comment]
@Dourak, is this still an issue in Cinnamon 4.2?

**Comment by Vahan86** [comment]
@icarter09, can we close the issue since there is no response from OP?

**Comment by weekens** [comment]
Hey guys! I think, this should be reopened, because in latest Cinnamon the multiload-indicator is still not working. I'm missing this badly in Mint. In other distributions like Ubuntu and Elementary this indicator works like a charm.

**Comment by mtwebster** [comment]
This seems to work fine for me.  Do you have the applet 'Xapp Status Applet' on the panel?  Try turning off indicator support (system settings->general), assuming it was activated, and use the xapp applet.

**Comment by weekens** [comment]
@mtwebster , no, this doesn't work for me. Sorry, didn't quite get the point about appindicator.

**Comment by mtwebster** [comment]
It even works with the standard systray applet *with* indicator support enabled.  Can you be more specific about how it doesn't work for you?  And provide your system info - mint version, etc.. it's possible something in your particular system prevents it from working properly.

If this switch is *off* the indicator should appear in the 'XApp Status Applet'.  With it *on*, it will show up in the traditional system tray.  The XApp status applet is *only* in Mint 19.3, so in previous versions you need to turn the switch on and use if from the status applet.

![ind](https://user-images.githubusercontent.com/262776/71479321-dbeaf000-27c1-11ea-9817-3f23c3c3cdaa.jpeg)

**Comment by weekens** [comment]
@mtwebster , I'm on Mint 19.3 (upgraded recently). I've installed XApp status applet and disabled indicators support just as you've described. After that, if I launch the `indicator-multiload`, all it's graphs are displayed "squashed" into a single icon:

![image](https://user-images.githubusercontent.com/790572/71541668-b1d63100-296d-11ea-84b8-67ddb8a409ce.png)

I tried to go with System Monitor applet, but it doesn't even display, showing API incompatibility error:

![image](https://user-images.githubusercontent.com/790572/71541690-ef3abe80-296d-11ea-8030-63990958c474.png)

(a popup hint for exclamation icon tells the applet contains Cinnamon API calls that are not supported anymore by the latest Cinnamon version).
